enterprise/admin/utils.py

```diff
@@ -111,4 +111,18 @@ def validate_email_to_link(email, raw_email=None, message_template=None, ignore_
         raise ValidationError(ValidationMessages.USER_ALREADY_REGISTERED.format(
             email=email, ec_name=existing_record.enterprise_customer.name
         ))
-    return bool(existing_record)
+    return existing_record or False
```

Comment:
[Here's a link to the relevant code.](https://github.com/open-craft/edx-enterprise/blob/8f4ec5dc1933639ad0496936ba7dc8193311f8a8/enterprise/admin/views.py#L170:L184). Basically, by actually returning the EnterpriseCustomerUser if it exists, we can use the extra data to make more relevant error messages - specifically in this case, we're using it to avoid causing errors where a linked user does exist already, but is attached to the EnterpriseCustomer we're operating on. Python objects are truthy/falsey, so this doesn't affect operation in any places where we're expecting a plain boolean object.